packages/xo-web/src/xo-app/backup-ng/file-restore/restore-file-modal.js

```diff
@@ -1,68 +1,95 @@
 import _ from 'intl'
 import ActionButton from 'action-button'
+import ButtonGroup from 'button-group'
 import Component from 'base-component'
-import endsWith from 'lodash/endsWith'
 import Icon from 'icon'
 import React from 'react'
-import replace from 'lodash/replace'
 import Select from 'form/select'
 import Tooltip from 'tooltip'
 import { Container, Col, Row } from 'grid'
 import { createSelector } from 'reselect'
 import { formatSize } from 'utils'
-import { filter, includes, isEmpty, map } from 'lodash'
+import {
+  endsWith,
+  filter,
+  find,
+  forEach,
+  includes,
+  isEmpty,
+  map,
+  replace,
+  startsWith,
+} from 'lodash'
 import { getRenderXoItemOfType } from 'render-xo-item'
 import { listPartitions, listFiles } from 'xo'
 
+// -----------------------------------------------------------------------------
+
 const BACKUP_RENDERER = getRenderXoItemOfType('backup')
 
+const diskOptionRenderer = disk => (
+  <span>
+    <Icon icon='disk' /> {disk.name}
+  </span>
+)
+
 const partitionOptionRenderer = partition => (
   <span>
     {partition.name} {partition.type}{' '}
     {partition.size && `(${formatSize(+partition.size)})`}
   </span>
 )
 
-const diskOptionRenderer = disk => <span>{disk.name}</span>
+const fileOptionRenderer = ({ isFile, name }) => (
+  <span>
+    <Icon icon={isFile ? 'file' : 'folder'} /> {name}
+  </span>
+)
 
-const fileOptionRenderer = file => <span>{file.name}</span>
+// -----------------------------------------------------------------------------
 
 const formatFilesOptions = (rawFiles, path) => {
   const files =
     path !== '/'
       ? [
           {
-            name: '..',
             id: '..',
-            path: getParentPath(path),
-            content: {},
+            isFile: false,
+            name: '..',
+            path: replace(path, /^(\/+.+)*(\/+.+)/, '$1/'),
```

Comment:
What does this replace do?

Reply:
I simply inlined `getParentPath` because I got rid of the other call of the function. I can put it back if you think it serves the code understandability.

It transforms a path to its parent folder path: `/path/to/folder/` → `/path/to/`.

Reply:
Maybe it would be best to keep the function to make the code clearer or use `path.unix.basename()`.

Otherwise, it's not immediately clear what it does.